### DISCUSSION TITLE
GECKO Model Adapter. Is it possible to define more than one compartment with the parameter 'obj.params.enzyme_comp' ?

### DISCUSSION THREAD

**Discussion by Zyklotron42** [SysBioChalmers/GECKO#386] [Q&A] [answered]
Hello there,

My question is about how to use the GECKO model adapter correctly. I don't really understand the parameter  'obj.params.enzyme_comp'. Neither in the GECKO protocol, nor anywhere else in the internet I find instructions what to do in the case that enzymes should be added in more than one compartment. Do I have to run the GECKO protocol multiple times with different ModelAdapters or is there any Syntax that allows me to define multiple compartments for the 'obj.params.enzyme_comp' at once? 

Thanks in advance!

**Comment by edkerk** [comment, marked as answer]
Thank you for reaching out, it appears that the instructions are indeed not sufficiently clear. The comparment name indicated in `obj.params.enzyme_comp` just determines where all enzyme/protein/protein pool pseudometabolites will be located. It actually doesn't matter which compartment that is, but one has to be selected, and it's been routine to select the cytosol. As a consequence, you end up with reactions like `metabolite A[m] + prot_XXXX[c] => metabolite[m]`, a mitochondrial reaction that uses a cytosolic protein. Strictly, one could have decided to place the protein pseudometabolites in different compartments, but it adds complexity on dealing with these models without having any benefit.
I'll clarify this in the tutorial and default model adapter file.

**Reply by edkerk** [reply]
Implemented in #385

**Reply by Zyklotron42** [reply]
Thank you so much that really helped a lot!
